src/os/filestore/FileStore.cc

```diff
@@ -4987,7 +4987,17 @@ int FileStore::list_collections(vector<coll_t>& ls, bool include_temp)
   }
 
   struct dirent *de = nullptr;
-  while ((de = ::readdir(dir))) {
+  while (true) {
+    errno = 0;
+    de = ::readdir(dir);
+    if (de == nullptr) {
+      if (errno != 0) {
+        r = -errno;
+        derr << "readdir failed " << fn << ": " << cpp_strerror(-r) << dendl;
```

Comment:
nit: I would prefer `cpp_strerror(r)`, but you can ignore (here and below).

Reply:
Hi, thanks for the review.

Actually I followed the style of other code in the file. So I'd like to keep this if it is ok.